Landing page

dingding-ye opened the pull request:
New elements:
- background created
- sliding panel created
- background color changeable
- rounded edges for sliding panel

Ovikx committed:
create src directory structure

dingding-ye committed:
new folders

joyjyu committed:
background

Ovikx committed:
remove unused import

dingding-ye committed:
updated panel import

dingding-ye committed:
Merge branch 'landing-page' of https://github.com/rice-apps/psy.cl-v1 into landing-page

dingding-ye committed:
working panel

dingding-ye committed:
min max height added

dingding-ye committed:
rounded edges

Ovikx committed:
add box shadow

kbonilla2 commented on lib/src/features/floating-panel/background.dart:
```diff
@@ -0,0 +1,14 @@
+  Widget build(BuildContext context) => Scaffold(
+        backgroundColor: Colors.red,
+        appBar: AppBar(
+            //title:Text(MyApp.title),
```
delete any unnecessary commented code

kbonilla2 commented on lib/main.dart:
```diff
@@ -44,72 +48,50 @@ class MyHomePage extends StatefulWidget {
+      ),
+      body: SlidingUpPanel(
+        panel: Center(
+          child: Text("This is the sliding Widget"),
```
are these texts needed for the functionality?

kbonilla2 commented on lib/src/widget/panel.dart:
```diff
@@ -0,0 +1,36 @@
+        padding: EdgeInsets.symmetric(horizontal: 24), 
+        child: Column(
+          crossAxisAlignment: CrossAxisAlignment.start,
+          // ignore: prefer_const_literals_to_create_immutables
```
unnecessary comments(?)

dingding-ye committed:
added scroll effect

dingding-ye committed:
Merge branch 'landing-page' of https://github.com/rice-apps/psy.cl-v1 into landing-page

Ovikx committed:
add PageView widget

Ovikx committed:
Merge branch 'landing-page' of https://github.com/rice-apps/psy.cl-v1 into landing-page

kbonilla2 reviewed (commented):
Overall this is a nice first pull request :) a few notes I have though are
- Remove any unnecessary comments/text widgets. This will make the codebase much cleaner, especially for everyone else that looks through your code for their updates.
- For the default text you guys have like "donuts yay" and the default Latin text, I would change it to "buttons here" or "data entry here" so the data entry group knows where to put their code on your codebase.
- For the floating panel, we would like it to go down about halfway/ 3/4ths down the screen and that's it. We don't want users to be able to drag down the panel all the way, just to the point under where the progress circle would be. Estimate how big the progress circle will be a set a temporary location for the floating panel to minimize to. Then we want users to be able to pull the floating panel to *almost* to the top where to can still see the top of the progress circle. Check the Figma for reference, I will also come around to explain this in person :)

dingding-ye committed:
added parallax

dingding-ye committed:
Merge branch 'landing-page' of https://github.com/rice-apps/psy.cl-v1 into landing-page

dingding-ye committed:
changed sliding panel height

dingding-ye committed:
edited background scrolling effect

Ovikx committed:
add page view indicator

Ovikx committed:
Merge branch 'landing-page' of https://github.com/rice-apps/psy.cl-v1 into landing-page

Ovikx committed:
move up page indicator

Ovikx committed:
add constants file

Ovikx committed:
add flexible page indicator positioning

Ovikx committed:
change page indicator effect and colors

Ovikx committed:
increase min panel height to what it's supposed to be

Ovikx committed:
refactor dir structure and clean up code

- move landing page features to src/features
- remove unnecessary comments
- add more meaningful text placeholders
- appease the linter
